pyanaconda/modules/payloads/source/harddrive/harddrive.py

```diff
@@ -43,8 +41,8 @@ def __init__(self):
         self._device = ""
         self.device_changed = Signal()
         self._install_tree_path = ""
-        self._device_mount = INSTALL_TREE + "_device"
-        self._iso_mount = INSTALL_TREE + "_iso"
+        self._device_mount = self._mount_point + "-device"
+        self._iso_mount = self._mount_point + "-iso"
```

Comment:
Maybe, you could replace `TearDownHardDriveSourceTask` with two instances of `TearDownMountTask`. It would be good to also use `SetUpMountTask` to check the mount points.

Reply:
The former is another facepalm moment for me. The latter, I am not sure. I'll try to remember and bring it up at the next opportunity.

Reply:
Haha, another facepalm - I finally got what you meant by the other half. Check implemented in-place now, discussion still needed. I don't see how a class that assumes exactly one of something can be subclassed easily into handling more than one of that. I'd need to override all of the ancestor code anyway.

Reply:
This solution seems to be fine.